### PR TITLE
Implement Session Model for Time Tracking Functionality in SwiftCuckoo

### LINKED ISSUE #2: Task Object Creation
**Description:** Create a `Task` object to hold time tracking data.
- **Goals:**
  - Define a `Task` struct or class with properties for task ID, start time, end time, and duration.

### PR DESCRIPTION
**Pull request by Tavernari**
#### Overview
This pull request introduces a new time tracking functionality to the `SwiftCuckoo` project by implementing a `Session` model that can encapsulate details about time-tracked tasks. The model includes properties for start time, optional end time, and a method to calculate the duration of the session. Additionally, comprehensive unit tests have been added to ensure the functionality's reliability.

#### Key Changes
1. **New `Session` Model**:
   - A `Session` structure has been added to represent individual time tracking sessions.
   - Contains properties:
     - `id`: A unique identifier for the session.
     - `startTime`: The time the session begins.
     - `endTime`: An optional property representing when the session ends.
   - Implements a method `duration()` that calculates the time elapsed between `startTime` and `endTime`.

2. **Error Handling**:
   - Custom error types have been introduced for handling specific conditions, such as:
     - Missing `endTime`.
     - `endTime` being earlier than `startTime`.

3. **Dependencies Overview**:
   - Updated `Package.swift` to include the `swift-testing` library, which enhances our ability to write tests for the project.

4. **Unit Tests for `Session` Model**:
   - Added comprehensive tests in `SessionTests.swift` that cover various scenarios, including:
     - Throwing an error when the `endTime` is nil.
     - Correct duration calculation when `endTime` is set.
     - Error handling when `endTime` is earlier than `startTime`.
     - Checking the duration when `startTime` equals `endTime`.

5. **Scheme Files**:
   - Two new Xcode scheme files for managing build settings effectively during development and testing.

#### Motivation
Introducing the `Session` model aligns with the project's goal of facilitating productivity tracking. By enabling users to track time spent on tasks effectively, we aim to provide valuable insights that can help improve work efficiency.

#### Testing
- The newly added tests ensure that the `Session` model behaves as expected under different conditions and edge cases.
- All tests are passing, confirming that the implementation is robust and meets the specified requirements.

#### Conclusion
This PR marks an important step forward in enhancing the functionality of the `SwiftCuckoo` project. The addition of the `Session` model, along with appropriate tests and error handling, will significantly benefit users looking to track their tasks and optimize their productivity. 

Please review the changes and provide any feedback!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
 # SwiftCuckoo
+![Build](https://github.com/tavernari/swiftcuckoo/actions/workflows/swift.yml/badge.svg)
+[![Maintainability](https://api.codeclimate.com/v1/badges/315d2f68bad03d358e42/maintainability)](https://codeclimate.com/github/Tavernari/SwiftCuckoo/maintainability)
```
Comment (Tavernari) [inline README.md:3]: ℹ️ Add badges to the README like this:

Screenshot 2024-09-03 at 05 55 05

File: Sources/SwiftCuckoo/Session.swift [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        guard let endTime else {
+            throw Error.missingEndTime
+        }
+        
+        guard endTime >= startTime else {
+            throw Error.endTimeBeforeStartTime
+        }
+        
+        return endTime.timeIntervalSince(startTime)
+    }
```
Comment (Tavernari) [inline Sources/SwiftCuckoo/Session.swift:56]: ℹ️ The idea of the duration throwing the error is for the code to explain why it is failing instead of just returning as `nil`

File: Sources/SwiftCuckoo/Session.swift [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        /// The raw string value representing the identifier.
+        public let rawValue: String
+        
+        /// Initializes a new identifier with a given raw value.
+        ///
+        /// - Parameter rawValue: The string value that represents the identifier.
+        public init(rawValue: String) {
+            self.rawValue = rawValue
+        }
+    }
```
Comment (Tavernari) [inline Sources/SwiftCuckoo/Session.swift:74]: ℹ️ This `Identifier` approach is similar to `Notification.Name`.

So instead of using `String` directly to identify a `Session` it uses this structure and the advantage of it is that the client can add auto-completion possibility.

## Example: 

### Without auto-completion:
```swift
let session = Session(id: Session.Identifier(rawValue: "test"), ...)
```

### with auto-completion
```swift
extension Session.Identifier {
    static let test = Session.Identifier(rawValue: "test")
}

let session = Session(id: .test, ...)
```

On the **SessionTests.swift** has an example:
Screenshot 2024-09-03 at 06 00 11


I like this approach because it avoids typo errors.

Comment (dhargrow) [reply Sources/SwiftCuckoo/Session.swift:74]: This is clever

**Review by dhargrow — APPROVED** [review]
